src/Oro/Bundle/DataGridBundle/Resources/public/js/datagrid/page-size.js

```diff
@@ -110,7 +110,7 @@ define([
                 _.bind(
                     function(item) {
                         return item.size === undefined ?
-                            this.collection.state.pageSize === item : this.collection.state.pageSize === item.size;
+                            this.collection.state.pageSize == item : this.collection.state.pageSize == item.size;
                     },
```

Comment:
@mbessolov, the issue is already fixed in 1.9.* and soon it will be in 1.8.3.

And, yes, for similar cases there have to be explicit type conversion.